src/server/config/config.js

```diff
@@ -163,8 +186,23 @@ module.exports = class Config {
 
   getSchema() {
     if (!this[schema]) {
-      let objKeys = zipObject([...this[schemaKeys]]);
-      this[schema] = Joi.object().keys(objKeys).default();
+      this[schema] = (function convertToSchema(children) {
+        let objKeys = Object.keys(children);
+        let schema = Joi.object().keys(objKeys).default();
```

Comment:
Yup, I read the Joi docs wrong before. 

The call to `.keys()` is required, otherwise the validation fails (and the tests do as a result), but it just needs to be given an empty object, since the actual keys are appended after the fact.